typings/app.d.ts

```diff
@@ -123,7 +123,7 @@ declare namespace firebase.auth {
     currentUser : firebase.User | null ;
     fetchProvidersForEmail (email : string ) : firebase.Promise < any > ;
     getRedirectResult ( ) : firebase.Promise < any > ;
-    onAuthStateChanged (nextOrObserver : Object , error ? : (a : firebase.auth.Error ) => any , completed ? : ( ) => any ) : ( ) => any ;
+    onAuthStateChanged (nextOrObserver : Object | ((user: firebase.User | null) => any), error ? : (a : firebase.auth.Error ) => any , completed ? : ( ) => any ) : ( ) => any ;
     onIdTokenChanged (nextOrObserver : Object , error ? : (a : firebase.auth.Error ) => any , completed ? : ( ) => any ) : ( ) => any ;
```

Comment:
Good point. I'll check if there's any other places too.

How about also introducing an `Observer` interface to use instead of the more generic `Object`?
```ts
interface Observer {
  next?: (user: firebase.User | null) => any;
  error?: (error: firebase.auth.Error) => any;
  complete?: () => any;
}
```